js/app/models/calendarListItemModel.js

```diff
@@ -49,7 +49,16 @@ app.factory('CalendarListItem', function($rootScope, $window, Calendar, WebCal,
 			},
 			publicSharingURL: {
 				get: () => {
-					return $rootScope.root + 'p/' + context.calendar.publicToken;
+					let displayname = context.calendar.displayname
+						.replace(/\s+/g, '-').replace(/[^\w\-]+/g, '')
+						.replace(/\-\-+/g, '-').replace(/^-+/, '')
+						.replace(/-+$/, '');
```

Comment:
@georgehrke Just out of curiosity. Couldn't you combine at least the combine the regex for '-' and '' with groups?